### PR TITLE
App local extensions folder, and infrastructure to deploy items from NuGet packages

### PR DESCRIPTION
**Pull request by dsplaisted**
- Add feature flag (enabled for .NET Core) to use Extensions subfolder of  MSBuild Exe directory as MSBuildExtensionsPath property
- Use NuGet v3 feed (this should be faster)
- Add target to copy contents folders from NuGet packages referenced by `src/.nuget/packages.config` to output folder, preserving the folder tree (which NuGet doesn't currently support)
- Add target to deploy .NET runtime and other dependencies to output folder, removed runtime sections from project.json files

### CODE REVIEW

File: src/XMakeTasks/Microsoft.Build.Tasks.csproj [hunk 1]
```diff
@@ -727,7 +727,7 @@
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
       <SubType>Designer</SubType>
     </Content>
-    <Content Include="Microsoft.Common.props">
+    <Content Include="Extensions\14.1\Microsoft.Common.props">
```
Comment (rainersigwald) [inline src/XMakeTasks/Microsoft.Build.Tasks.csproj]: Why include version in the source tree here?  Seems maintenance-prone.

File: src/dir.targets [hunk 2]
```diff
@@ -2,7 +2,9 @@
   <Import Project="..\dir.targets" />
 
   <PropertyGroup>
-    <CopyNuGetImplementations Condition="'$(IsTestProject)' == 'true'">true</CopyNuGetImplementations>
+    <!-- Don't copy NuGet implementations directly, since that would require setting a runtime and targeting a specific architecture.
+         Instead, it they will be deployed by DeployDependencies.proj. -->
```
Comment (rainersigwald) [inline src/dir.targets]: Hooray for comment :+1:

File: targets/DeployDependencies.proj [hunk 3]
```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project DefaultTargets="DeployDependencies" ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="..\dir.props" />
+
+  <Target Name="CopyPackageContent">
+    <ItemGroup>
+      <NuspecFilesInPackagesDir Include="$(PackagesDir)\**\*.nupkg" />
+      <PackagesFoldersInPackagesDir Include="@(NuspecFilesInPackagesDir->'%(RootDir)%(Directory)')" />
+    </ItemGroup>
+    <Msbuild Targets="CopyExtensionsFromPackage"
```
Comment (rainersigwald) [inline targets/DeployDependencies.proj]: Style/capitalization: `MSBuild`

**Comment by rainersigwald** [comment]
My local build is failing for the same reason as the CI: packages not resolved:
```
  Unable to resolve Microsoft.Net.Compilers.NetCore (= 0.1.0-dev) for DNXCore,Version=v5.0. (TaskId:227)
```

In verbose mode it seems to be looking at the right sources, though:

```
Restoring packages for O:\msbuild\targets\runtime.project.json...
Using source C:\Users\raines.NTDEV\AppData\Local\NuGet\Cache.
Using source https://www.myget.org/F/dotnet-core/.
Using source https://www.myget.org/F/dotnet-corefx/.
Using source https://www.myget.org/F/dotnet-corefxtestdata/.
Using source https://www.myget.org/F/dotnet-buildtools/.
Using source https://api.nuget.org/v3/index.json.
```

Is this something local to your machine?

File: targets/DeployDependencies.proj [hunk 4]
```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project DefaultTargets="DeployDependencies" ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="..\dir.props" />
+
+  <Target Name="CopyPackageContent">
```
Comment (rainersigwald) [inline targets/DeployDependencies.proj]: I think you can avoid this double-hop indirection.  I sent you https://github.com/dsplaisted/msbuild/pull/1.

File: src/XMakeTasks/Microsoft.Build.Tasks.csproj (hunk 1 above)
Comment (dsplaisted) [inline src/XMakeTasks/Microsoft.Build.Tasks.csproj]: @rainersigwald 

> Why include version in the source tree here? Seems maintenance-prone.

I've filed #256 to clean this up.

**Comment by rainersigwald** [comment]
LGTM

**Pull request merged**